libs/com.etendorx.utils.auth/src/main/java/com/etendorx/utils/auth/key/context/FilterContext.java

```diff
@@ -67,45 +63,44 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
           .filter(uri -> uri.isAllowed(request.getRequestURI()))
           .findFirst();
       if (optionalAllowedUri.isEmpty()) {
-        throw new ForbiddenException(request.getRequestURI() + " is not allowed for this credentials");
+        throw new ForbiddenException(
+            request.getRequestURI() + " is not allowed for this credentials");
       }
     }
     AppContext.setCurrentUser(userContext);
     filterChain.doFilter(request, response);
   }
 
-  public void setUserContextFromToken(String token, HttpServletRequest request) {
+  public void setUserContextFromToken(String publicKey, String token, HttpServletRequest request) {
     setUserContextFromToken(userContext, publicKey, token, request);
   }
 
-  public static void setUserContextFromToken(UserContext userContext, String publicKey, String token, HttpServletRequest req) {
+  public static void setUserContextFromToken(UserContext userContext, String publicKey,
+      String token, HttpServletRequest req) {
+    TokenUtil.convertToken(userContext, publicKey, token);
     String noActiveFilterParameter = req.getParameter(NO_ACTIVE_FILTER_PARAMETER);
     String triggerEnabledParam = req.getParameter(TRIGGER_ENABLED_PARAMETER);
     String dateFormatParam = req.getParameter(DATE_FORMAT_PARAMETER);
     String timeZoneParam = req.getParameter(TIME_ZONE_PARAMETER);
     String restMethod = req.getMethod();
-    Map<String, Object> tokenValuesMap = ContextUtils.getTokenValues(publicKey, token);
-    userContext.setUserId((String) tokenValuesMap.get(JwtKeyUtils.USER_ID_CLAIM));
-    userContext.setClientId((String) tokenValuesMap.get(JwtKeyUtils.CLIENT_ID_CLAIM));
-    userContext.setOrganizationId((String) tokenValuesMap.get(JwtKeyUtils.ORG_ID));
-    userContext.setRoleId((String) tokenValuesMap.get(JwtKeyUtils.ROLE_ID));
-    userContext.setSearchKey((String) tokenValuesMap.get(JwtKeyUtils.SERVICE_SEARCH_KEY));
-    userContext.setServiceId((String) tokenValuesMap.get(JwtKeyUtils.SERVICE_ID));
-    boolean noActiveFilter = !parseBooleanParameter(noActiveFilterParameter, NO_ACTIVE_FILTER_PARAMETER, false);
+    boolean noActiveFilter = !parseBooleanParameter(noActiveFilterParameter,
+        NO_ACTIVE_FILTER_PARAMETER, false);
     userContext.setActive(noActiveFilter);
     userContext.setAuthToken(token);
     userContext.setRestMethod(restMethod);
-    boolean isTriggerEnabled = parseBooleanParameter(triggerEnabledParam, TRIGGER_ENABLED_PARAMETER, true);
+    boolean isTriggerEnabled = parseBooleanParameter(triggerEnabledParam, TRIGGER_ENABLED_PARAMETER,
+        true);
     userContext.setTriggerEnabled(isTriggerEnabled);
-    if(StringUtils.isNotBlank(dateFormatParam)) {
+    if (StringUtils.isNotBlank(dateFormatParam)) {
       userContext.setDateFormat(dateFormatParam);
     }
-    if(StringUtils.isNotBlank(dateFormatParam)) {
+    if (StringUtils.isNotBlank(dateFormatParam)) {
       userContext.setTimeZone(timeZoneParam);
     }
```

Comment:
**Suggestion:** The setUserContextFromToken method is checking for the presence of certain parameters and setting them in the UserContext object. Consider extracting this logic into a separate method for better readability.
```suggestion
  setRequestParametersInUserContext(req, userContext, token);
```